lib/server/index.js

```diff
@@ -217,10 +217,17 @@ class Server extends EventEmitter {
       this.proxy = new httpProxy.createProxyServer({changeOrigin: true});
 
       this.proxy.on('error', (err, req, res) => {
-        res.status(500).json(err);
+        res?.status(500).json(err);
       });
 
-      Object.keys(proxies).forEach(url => {
+      Object.keys(proxies).forEach((url) => {
+        if (proxies[url].ws === true) {
```

Comment:
Would it be possible to add a test for this? Here are the existing ones https://github.com/testem/testem/blob/master/tests/server_tests.js